.github/workflows/codecov.yml

```diff
@@ -0,0 +1,134 @@
+name: Code Coverage
+on: # only on merges to main
+  push:
+    branches:
+      - main
+concurrency:
+  group: format('{0}-{1}', ${{ github.ref }}, 'Code Coverage')
+  cancel-in-progress: true
+
+permissions: read-all
+
+env:
+  GITHUB_PR_HEAD_SHA: "${{ github.event.pull_request.head.sha }}"
```

Comment:
I am unsure if this is correct as the event you want to run this workflow on is `push.branches: main`, the event will never be `pull_request`.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

Reply:
Yeah, if we end up saying that we only care about `main` and choose to forgo the PR comments noting its impact on code coverage then I would remove that (and likely some other parts) from the workflow yaml.

Reply:
I'll make these changes now, since the workflow as-is is main only. I can add additional handling later if we decide that we want to run it on PRs as well.